fdbserver/MockPeekCursor.h

```diff
@@ -21,39 +21,102 @@
 #ifndef FDBSERVER_MOCK_PEEK_CURSOR
 #define FDBSERVER_MOCK_PEEK_CURSOR
 
+#include <ostream>
 #include "fdbserver/LogSystem.h"
 
+struct VersionedMessage {
+	// Every message should be able to have independent arenas so that they can be freed separately after consumed.
+	Arena arena;
+	Version version;
+	// Subversion is not used for now.
+	uint32_t sub = 0;
+	StringRef message;
+	VectorRef<Tag> tags{};
+	VersionedMessage() = default;
+	VersionedMessage(const Arena& arena, Version version, const StringRef& message, const VectorRef<Tag>& tags)
+	  : arena(arena), version(version), message(message), tags(tags) {}
+
+	friend std::ostream& operator<<(std::ostream& os, const VersionedMessage& versionedMessage) {
+		os << "VersionedMessage [version: " << versionedMessage.version << " sub: " << versionedMessage.sub
+		   << " message: " << StringRef(versionedMessage.message).toHexString() << " tags: ";
+		for (auto tag : versionedMessage.tags) {
+			os << tag.toString() << " ";
+		}
+		os << "]";
+		return os;
+	}
+};
+
+struct VersionedMessageSupplier final : ReferenceCounted<VersionedMessageSupplier> {
+	int i = 0;
+	int end;
+	VectorRef<Tag> tags{};
+	int advanceVersionsPerMutation;
+
+	Optional<VersionedMessage> get() {
+		// std::cout << "OnDemandVersionedMessageSupplier get " << i << ", end " << end << std::endl;
+		ASSERT(i <= end);
+		if (i == end) {
+			return Optional<VersionedMessage>();
+		}
+		Arena arena;
+		MutationRef mutation(arena,
+		                     MutationRef::SetValue,
+		                     StringRef(arena, "Key-" + std::to_string(i)),
```

Comment:
Please add a TODO to make these keys more random. The performance of keys inserted in order is vastly different than the performance of keys inserted out of order.